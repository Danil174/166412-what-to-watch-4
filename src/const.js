export const TextMovieRatings = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`,
};

export const SmallCardVideoSettings = {
  WIDTH: `100%`,
  HEIGHT: `100%`,
  IS_MUTED: true,
};

export const DEFAULT_GENRE = `All genres`;
export const MAX_GENRES_LENGTH = 9;
export const INITTIAL_LIST_LENGTH = 8;
export const SIMILAR_LIST_LENGTH = 4;

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const smallCardVideoStartTimeOut = 1000;

export const LoadErrorsTexts = {
  LIST_FAIL: `Не удалось загрузить список фильмов.`,
  PROMO_FAIL: `Не удалось загрузить промо фильм.`,
};

export const AppRoute = {
  LOGIN: `/login`,
  MOVIE_PAGE: `/movie_page`,
  FAVORITES: `/favorites`,
  ROOT: `/`,
};
