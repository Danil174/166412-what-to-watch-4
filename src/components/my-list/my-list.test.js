import React from "react";
import renderer from "react-test-renderer";
import {Router} from "react-router-dom";
import history from "../../history.js";
import UserList from "./my-list.jsx";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {AuthorizationStatus} from "../../const.js";
import NameSpace from "../../reducer/name-space.js";

const mockStore = configureStore([]);

describe(`UserListTest`, () => {
  it(`render UserList`, () => {
    const store = mockStore({
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.NO_AUTH,
        userData: {avatarUrl: `test.jpg`},
      },
    });
    const tree = renderer
      .create(
          <Router history={history}>
            <Provider store={store}>
              <UserList />
            </Provider>
          </Router>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
