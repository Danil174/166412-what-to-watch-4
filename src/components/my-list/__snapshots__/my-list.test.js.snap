// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyListTest render MyList 1`] = `
<section
  className="loading"
>
  <div
    className="page-content"
  >
    <h1
      className="loading__title"
    >
      LOADING
    </h1>
    <div
      className="preload-wrapp"
    >
      <div
        className="dot"
      />
      <div
        className="dot"
      />
      <div
        className="dot"
      />
      <div
        className="dot"
      />
    </div>
  </div>
</section>
`;
